### DISCUSSION TITLE
Cutting edge CSS productivity

### DISCUSSION THREAD

**Discussion by matschik** [tailwindlabs/tailwindcss#3211] [Kind words]
I was not really convinced at first when I looked at the HTML with many CSS classes. But after using it for a little side project, just wow, I was so much productive: 
- no more BEM and all those underscores, 
- no more looking for the best CSS class name to distinguish them from others,
-  no worry about class scope availability, 
- no more searching through a long CSS stylesheet while leaving my HTML structure,...

Intuitivity and productivity and all of this with better performance (purgeCSS).
Even more productivity with TailwindUI giving me a lot of UI/UX inspiration, and quality with SVG icons from heroicons.
You not only introduced another CSS framework, but a new and intuitive way to use and write CSS. I even learn some things about CSS through really well made TailwindCSS documentation.

I'm using it for all my side-projects and also for a big project coming very soon so i'm using it every day !
It's always hard to convince people to use TailwindCSS primarily because of old habits, but I know that, if you don't try it, you will never fully understand the benefits.

So, Thank you soooo much Adam, Steve and all the other contributors to these projects ! <3
Your hard work is making developer life easier to create faster amazing projects !!

**Comment by simonswiss** [comment]
Amazing, thank you @matschik ❤️

It's indeed hard to convince people to try it (and like it!) at first, but... there is a pretty common pattern of folks going through the exact same steps than you described here.

Thanks again for sharing your experience!

**Comment by Prosacc** [comment]
I spent a lot of time thinking about how to optimally configure this process. If you're intentional about it you can get some impressive productivity gains in all areas of the process. Here's 8 ways to do that, from the beginning of the project through to the end:

Start from a solid base
Agree on design shortcuts
Use handoff tools
Use a CSS framework
Re-use your components
Use Emmet
Use Polypane
Set up automated quality checking




[upsers](https://www.upsers.online/)

**Comment by Williamssonn** [comment]
progress in web design techniques shows no signs of slowing down. As the proponents of the early web are staggering around with thousand-yard stares, you can't help but feel a little over-awed by the plethora of techniques and frameworks available nowadays. 

People are championing the new approaches that are about to change the way we work once again, such as Flexbox and CSS Grid. Rachel Andrew and Jen Simmons have been encouraging developers to look at these technologies now, in readiness for their arrival.

[iMessage Download](https://www.imessageapp.org/)
